### PR TITLE
Fix apiserver init check

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
This replaces the `etcdctl endpoint health` with a `etcdctl put quorum-check something`.

The difference is that `etcdctl endpoint health` checks if all specified endpoints are healthy - in our case 3. This can lead to downtime of the API server if only 1 etcd is down.

`etcdctl put quorum-check something` Does a simple write to etcd. If it succeeds we have a quorum and know that the API server can start.

**Release note**:
```release-note
Ensure that the API server init container only waits for etcd quorum and not all members.
```

### CODE REVIEW

File: api/pkg/resources/apiserver/deployment.go [hunk 1]
```diff
@@ -109,7 +109,8 @@ func DeploymentCreator(data resources.DeploymentDataProvider) resources.Deployme
 				Command: []string{
 					"/bin/sh",
 					"-ec",
-					fmt.Sprintf("until ETCDCTL_API=3 /usr/local/bin/etcdctl --cacert=/etc/etcd/pki/client/ca.crt --cert=/etc/etcd/pki/client/apiserver-etcd-client.crt --key=/etc/etcd/pki/client/apiserver-etcd-client.key --dial-timeout=2s --endpoints='%s' endpoint health; do echo waiting for etcd; sleep 2; done;", strings.Join(etcdEndpoints, ",")),
+					// Write a key to etcd. If we have quorum it will succeed.
+					fmt.Sprintf("until ETCDCTL_API=3 /usr/local/bin/etcdctl --cacert=/etc/etcd/pki/client/ca.crt --cert=/etc/etcd/pki/client/apiserver-etcd-client.crt --key=/etc/etcd/pki/client/apiserver-etcd-client.key --dial-timeout=2s --endpoints='%s' put quorum-check something; do echo waiting for etcd; sleep 2; done;", strings.Join(etcdEndpoints, ",")),
```
Comment (alvaroaleman) [inline api/pkg/resources/apiserver/deployment.go]: Can we maybe prefix this with `kubermatic/` so we have one defined keyspace?

Comment (mrIncompetent) [reply api/pkg/resources/apiserver/deployment.go]: Done

**Pull request merged**